common/math/geometry/src/polygon/line_segment.cpp

```diff
@@ -241,57 +209,37 @@ auto LineSegment::getIntersection2DSValue(const LineSegment & line, const bool d
 auto LineSegment::getIntersection2D(const LineSegment & line) const
   -> std::optional<geometry_msgs::msg::Point>
 {
-  const auto s = getIntersection2DSValue(line, false);
-  return s ? geometry_msgs::build<geometry_msgs::msg::Point>()
-               .x(s.value() * start_point.x + (1.0 - s.value()) * end_point.x)
-               .y(s.value() * start_point.y + (1.0 - s.value()) * end_point.y)
-               .z(s.value() * start_point.z + (1.0 - s.value()) * end_point.z)
-           : std::optional<geometry_msgs::msg::Point>();
+  return math::geometry::getIntersection2D(*this, line);
 }
 
-auto LineSegment::getVector() const -> geometry_msgs::msg::Vector3
-{
-  using math::geometry::operator-;
-  const auto result_pt = end_point - start_point;
-  auto result = geometry_msgs::msg::Vector3();
-  result.x = result_pt.x;
-  result.y = result_pt.y;
-  result.z = result_pt.z;
-  return result;
-}
+auto LineSegment::getVector() const -> const geometry_msgs::msg::Vector3 & { return vector; }
 
 /**
  * @brief Get normal vector of the line segment.
  * @return geometry_msgs::msg::Vector3 
  */
 auto LineSegment::getNormalVector() const -> geometry_msgs::msg::Vector3
 {
-  geometry_msgs::msg::Vector3 tangent_vec = getVector();
-  double theta = M_PI / 2.0;
+  const double theta = M_PI / 2.0;
   return geometry_msgs::build<geometry_msgs::msg::Vector3>()
-    .x(tangent_vec.x * std::cos(theta) - tangent_vec.y * std::sin(theta))
-    .y(tangent_vec.x * std::sin(theta) + tangent_vec.y * std::cos(theta))
-    .z(0);
+    .x(getVector().x * std::cos(theta) - getVector().y * std::sin(theta))
+    .y(getVector().x * std::sin(theta) + getVector().y * std::cos(theta))
+    .z(0.0);
 }
 
-auto LineSegment::get2DVector() const -> geometry_msgs::msg::Vector3
-{
-  return geometry_msgs::build<geometry_msgs::msg::Vector3>()
-    .x(end_point.x - start_point.x)
-    .y(end_point.y - start_point.y)
-    .z(0);
-}
+auto LineSegment::get2DVector() const -> const geometry_msgs::msg::Vector3 & { return vector_2d; }
```

Comment:
Member variable `vector_2d` is const variables and cannot be modified outside the member initializer list, so they can be public rather than private.
Getters are unnecessary for public and const members.